packages/create-astro/src/actions/typescript.ts

```diff
@@ -107,8 +114,19 @@ const FILES_TO_UPDATE = {
 			if (typeof buildScript === 'string' && !buildScript.includes('astro check')) {
 				// Mutate the existing object to avoid changing user-defined script order
 				parsedPackageJson.scripts.build = `astro check && ${buildScript}`;
-				await writeFile(file, JSON.stringify(parsedPackageJson, null, indent), 'utf-8');
 			}
+
+			if (!options.ctx.install || !installSucceeded) {
+				const [astroCheckVersion, typescriptVersion] = await Promise.all([
+					getVersion(options.ctx.packageManager, '@astrojs/check'),
+					getVersion(options.ctx.packageManager, 'typescript'),
+				]);
+				parsedPackageJson.dependencies ??= {}
+				parsedPackageJson.dependencies['@astrojs/check'] = astroCheckVersion;
+				parsedPackageJson.dependencies.typescript = typescriptVersion;
```

Comment:
We probably want to use a `^` instead of pinning an exact version
```suggestion
				parsedPackageJson.dependencies['@astrojs/check'] = `^${astroCheckVersion}`;
				parsedPackageJson.dependencies.typescript = `^${typescriptVersion}`;
```

Reply:
I was not sure about the format of `getVersion` return 👍